frontend2/src/views/Home.tsx

```diff
@@ -9,32 +9,56 @@ import TeamChart, {
   type ChartData,
 } from "../components/tables/chart/TeamChart";
 import { useQueryClient } from "@tanstack/react-query";
-import { useTopRatingHistoryList } from "api/compete/useCompete";
+import {
+  useTopRatingHistoryList,
+  useUserRatingHistoryList,
+} from "api/compete/useCompete";
+import ScrimmagingRecord from "components/compete/ScrimmagingRecord";
+import { useUserTeam } from "api/team/useTeam";
 
 const Home: React.FC = () => {
   const { episodeId } = useEpisodeId();
   const queryClient = useQueryClient();
   const episode = useEpisodeInfo({ id: episodeId });
   const nextTournament = useNextTournament({ episodeId });
   const topRatingHistory = useTopRatingHistoryList({ episodeId }, queryClient);
+  const userTeam = useUserTeam({ episodeId });
+  const userTeamRatingHistory = useUserRatingHistoryList({ episodeId });
 
   const SOCIAL =
     "hover:drop-shadow-lg hover:opacity-80 transition-opacity duration-300 ease-in-out";
 
-  const ratingData: Record<string, ChartData[]> | undefined = useMemo(() => {
-    if (!topRatingHistory.isSuccess) return undefined;
-    const ratingRecord: Record<string, ChartData[]> = {};
-    return topRatingHistory.data.reduce((record, teamData) => {
-      if (teamData.team_rating !== undefined) {
-        record[teamData.team_rating.team.name] =
-          teamData.team_rating.rating_history.map((match) => [
-            match.timestamp.getTime(),
-            match.rating,
-          ]);
-      }
-      return record;
-    }, ratingRecord);
-  }, [topRatingHistory]);
+  const userTeamRatingData: Record<string, ChartData[]> | undefined =
+    useMemo(() => {
+      if (!userTeamRatingHistory.isSuccess) return undefined;
+      const ratingRecord: Record<string, ChartData[]> = {};
+      return userTeamRatingHistory.data.reduce((record, teamData) => {
+        if (teamData.team_rating !== undefined) {
+          record[teamData.team_rating.team.name] =
+            teamData.team_rating.rating_history.map((match) => [
+              match.timestamp.getTime(),
+              match.rating,
+            ]);
+        }
+        return record;
+      }, ratingRecord);
+    }, [userTeamRatingHistory]);
+
+  const topTeamRatingData: Record<string, ChartData[]> | undefined =
+    useMemo(() => {
+      if (!topRatingHistory.isSuccess) return undefined;
+      const ratingRecord: Record<string, ChartData[]> = {};
+      return topRatingHistory.data.reduce((record, teamData) => {
+        if (teamData.team_rating !== undefined) {
+          record[teamData.team_rating.team.name] =
+            teamData.team_rating.rating_history.map((match) => [
+              match.timestamp.getTime(),
+              match.rating,
+            ]);
+        }
+        return record;
+      }, ratingRecord);
+    }, [topRatingHistory]);
```

Comment:
Lets see if we can improve the repetitiveness of this code. We can probably drop the stuff inside the useMemo into a function that takes a `rating history`

Reply:
Will do in #807 :rocket: